modules/core/shared/src/main/scala/eu/timepit/refined/string.scala

```diff
@@ -40,14 +43,34 @@ object string extends StringValidate with StringInference {
   final case class XPath()
 }
 
-private[refined] trait StringValidate {
+object Predicates {
+  object IPv4 {
```

Comment:
Could you move the `IPv4` object into the `string` object right below the `IPv4` case class and remove the `Predicates` object? It makes sense to put auxiliary code of predicates into their proper companion.